cpp/src/arrow/util/simd.h

```diff
@@ -27,18 +27,25 @@
 #else
 // gcc/clang (possibly others)
 
-#  if defined(ARROW_HAVE_BMI2)
+#  if defined(ARROW_HAVE_BMI2) || defined(ARROW_HAVE_RUNTIME_BMI2)
```

Comment:
I try to find some common place to put simd specific code and found this one. However it seems this file is only adapting the static simd optimizations, thus the additions of the `ARROW_HAVE_RUNTIME*` which are for dynamic simd optimizations.